### PR TITLE
fix: agent install instructions

### PR DESCRIPTION
**Pull request by moshloop**

### CODE REVIEW

**Comment by moshloop** [comment]
@mainawycliffe - the schedule is not pulling through, can we just remove it ?

**Comment by moshloop** [comment]
@mainawycliffe there are sill differences in the values - 

Can we create a Helm Component like https://github.com/flanksource/docs/blob/main/common/src/components/Helm.jsx - the values are dynamic and single-sourced

**Comment by mainawycliffe** [comment]
@moshloop The current component is dynamic; I had something similar originally, but I found the template syntax to lead to more issues compared to Handlesbar for both the Yaml and CLI (you would end up need handle all nullable options, which meant a small change could make the template incorrect), they take in the same data object, but return a string based on Yaml handlebar template or the command to run for the CLI, I think the handlebar template for the Yaml and CLI command passing in data object for replacement is far much easier compared to using the template syntax. Other than that, both components work similarly.

File: src/components/Agents/InstalAgentInstruction/CLIInstallAgent.tsx [hunk 1]
```diff
@@ -24,53 +21,21 @@ helm install mc-agent flanksource/mission-control-agent -n "mission-control-agen
 
 {{#if kubeOptions}}
 helm install mc-agent-kubernetes flanksource/mission-control-kubernetes -n "mission-control-agent"  \\
-  --set scraper.clusterName="{{agentFormValues.name}}" \\
-  --set scraper.interval="{{kubeOptions.interval}}"
+  --set clusterName="{{agentFormValues.name}}" \\
+  --set scraper.schedule="{{kubeOptions.schedule}}"
```
Comment (moshloop) [inline src/components/Agents/InstalAgentInstruction/CLIInstallAgent.tsx:25]: these should be properties passed into a HelmChart component that then selectively renders the flux or helm cli commands. What are we doing for the registry charts ?

File: src/ui/HelmSnippet/FluxInstallAgent.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+{{/if}}
+{{#if createRepository}}
+apiVersion: source.toolkit.fluxcd.io/v1beta1
+kind: HelmRepository
+metadata:
+  name: {{ repoName }}
+  namespace: {{ namespace }}
+spec:
+  interval: 5m0s
+  url: https://flanksource.github.io/charts
```
Comment (moshloop) [inline src/ui/HelmSnippet/FluxInstallAgent.tsx:23]: this should be configurable with a default

File: src/ui/HelmSnippet/CLIInstallAgent.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+EOF
+
+
+{{/if}}
+{{#if createRepo }}
+helm repo add {{ chart }} {{ repoName }}/{{ chart }}
+{{/if}}
+helm repo update
+
+helm install mc-agent flanksource/mission-control-agent -n "{{{ namespace }}}" \\
```
Comment (moshloop) [inline src/ui/HelmSnippet/CLIInstallAgent.tsx:20]: chart name and source should be configurable

File: src/components/Agents/InstalAgentInstruction/InstallAgentModal.tsx [hunk 4]
```diff
@@ -96,7 +83,77 @@ export default function InstallAgentModal({
...
+                }
+              ]
+            : [])
+        ]
+      },
+      ...(kubeOptions?.enabled
+        ? [
+            {
+              chart: "mc-agent-kubernetes",
+              namespace: "mc-agent-kubernetes",
```
Comment (moshloop) [inline src/components/Agents/InstalAgentInstruction/InstallAgentModal.tsx:140]: The namespace needs to match above

File: src/ui/HelmSnippet/CLISnippet.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+{{/if}}
+{{#if this.createRepo }}
+helm repo add {{{ this.chart }}} {{{ this.repoName }}}/{{{ this.chart }}}
+{{/if}}
+{{#if this.updateHelmRepo }}
+helm repo update
+{{/if}}
+
+helm install mc-agent {{{ this.repoName }}}/{{{ this.chart }}} -n "{{{ this.namespace }}}" \\
```
Comment (moshloop) [inline src/ui/HelmSnippet/CLISnippet.tsx:22]: mc-agent should be configurable

File: src/ui/HelmSnippet/CLISnippet.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+      {
+        charts: data
+      },
+      {}
+    );
+  }, [data]);
+
+  return (
+    <div className="flex flex-1 flex-col gap-4 overflow-y-auto p-2">
+      <p>Copy the following command to install agent</p>
```
Comment (moshloop) [inline src/ui/HelmSnippet/CLISnippet.tsx:57]:
```suggestion
      <p>Copy the following command to install the chart</p>
```

File: src/ui/HelmSnippet/__tests__/__snapshots__/CLISnippet.unit.test.tsx.snap [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+"helm repo add mission-control-agent flanksource/mission-control-agent
+
+helm install mc-agent flanksource/mission-control-agent -n "mission-control-agent" \\
+  --set upstream.createSecret=true  \\ 
+  --set upstream.host=http://localhost:3000  \\ 
+  --set upstream.username=token  \\ 
+  --set upstream.password=password  \\ 
+  --set upstream.agentName=test-new-agent-instructions  \\ 
+  --set pushTelemetry.enabled=true  \\ 
+  --set pushTelemetry.topologyName=https://incident-commander.demo.aws.flanksource.com-test-new-agent-instructions  \\
```
Comment (moshloop) [inline src/ui/HelmSnippet/__tests__/__snapshots__/CLISnippet.unit.test.tsx.snap:13]:
```suggestion
  --set pushTelemetry.topologyName=incident-commander.demo.aws-test-new-agent-instructions  \\
```

File: src/ui/HelmSnippet/__tests__/__snapshots__/CLISnippet.unit.test.tsx.snap [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  --create-namespace 
+   
+
+helm install mc-agent flanksource/mc-agent-kubernetes -n "mission-control-agent" \\
```
Comment (moshloop) [inline src/ui/HelmSnippet/__tests__/__snapshots__/CLISnippet.unit.test.tsx.snap:33]:
```suggestion
helm install mc-kubernetes flanksource/mission-control-kubernetes -n "mission-control-agent" \\
```

File: src/ui/HelmSnippet/CLISnippet.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+// kubernetes agent if the user has enabled it.
+const helmCommand = `{{#each charts}}
+{{#if this.valueFile }}
+cat > values.yaml << EOF
+{{ this.valueFile }}
+EOF
+
+{{/if}}
+{{#if this.createRepo }}
+helm repo add {{{ this.chart }}} {{{ this.repoName }}}/{{{ this.chart }}}
```
Comment (moshloop) [inline src/ui/HelmSnippet/CLISnippet.tsx:16]:
```suggestion
helm repo add {{{ this.repoName }}} {{{ this.chartUrl }}}
```

File: src/ui/HelmSnippet/CLISnippet.tsx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+{{/if}}
+{{#if this.updateHelmRepo }}
+helm repo update
+{{/if}}
+
+helm install {{{ this.releaseName }}} {{{ this.repoName }}}/{{{ this.chart }}} -n "{{{ this.namespace }}}" \\
```
Comment (moshloop) [inline src/ui/HelmSnippet/CLISnippet.tsx:22]:
```suggestion
helm install {{{ this.chart }}} {{{ this.repoName }}}/{{{ this.chart }}} -n "{{{ this.namespace }}}" \\
```

File: src/ui/HelmSnippet/FluxSnippet.tsx [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+kind: HelmRepository
+metadata:
+  name: {{ this.repoName }}
+  namespace: {{ this.namespace }}
+spec:
+  interval: 5m0s
+  url: {{{ this.chartUrl }}}
+---
+{{/if}}
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
```
Comment (moshloop) [inline src/ui/HelmSnippet/FluxSnippet.tsx:27]:
```suggestion
apiVersion: helm.toolkit.fluxcd.io/v2
```

File: src/ui/HelmSnippet/FluxSnippet.tsx [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+const fluxTemplate = `{{#each charts }}
+{{#if this.createNamespace}}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name:  {{ this.namespace }}
+---
+{{/if}}
+{{#if this.createRepo}}
+apiVersion: source.toolkit.fluxcd.io/v1beta1
```
Comment (moshloop) [inline src/ui/HelmSnippet/FluxSnippet.tsx:17]:
```suggestion
apiVersion: source.toolkit.fluxcd.io/v1
```

**Pull request merged**